micrometer-observation/src/main/java/io/micrometer/observation/Observations.java

```diff
@@ -61,6 +61,8 @@ public static ObservationRegistry getGlobalRegistry() {
         return globalRegistry;
     }
 
+    // We know delegate.get() cannot be null but NullAway does not.
+    @SuppressWarnings("NullAway")
```

Comment:
Is this a bug we should report?
Also, can `@SuppressWarnings` go on the `delegate`?

Reply:
AtomicReference is a JDK class, so we can't directly control its specified internal nullability. NullAway is considering that `AtomicReference.get` returns a nullable value, it seems. Ideally, by not marking the generic type as `@Nullable` we'd hope it is treated as `@NonNull` due to the package containing the field being annotated `@NullMarked`. It might be related to https://github.com/uber/NullAway/issues/1161, but I'm not sure.

Moving the suppression to the field doesn't help because the compiler error is thrown from the `delegate.get` call sites.